plugins/nexus-repository-manager/src/api/nexus-repository-manager-api-client/nexus-repository-manager-api-client.ts

```diff
@@ -99,13 +117,20 @@ export class NexusRepositoryManagerApiClient
 
     const path = /\/(repository\/.*)/.exec(url)?.at(1);
 
-    return (await this.fetcher(`${proxyUrl}/${path}`)) as RawAsset;
+    return (await this.fetcher(
+      `${proxyUrl}/${path}`,
+      additionalHeaders,
+    )) as RawAsset;
   }
 
   private async getRawAssets(component: ComponentXO) {
     const assets = await Promise.all(
       component.assets?.map(
-        async asset => await this.getRawAsset(asset.downloadUrl),
+        async asset =>
+          await this.getRawAsset(
+            asset.downloadUrl,
+            component.format === 'docker' ? DOCKER_MANIFEST_HEADERS : {},
+          ),
```

Comment:
Sadly I cannot make this a git suggestion; however, I would rewrite `getRawAssets` with a header helper function as the following.

```ts
function getAdditionalHeaders(format?: string): HeadersInit {
  switch (format /* NOSONAR - use switch for expandability */) {
    case 'docker':
      return DOCKER_MANIFEST_HEADERS;
    default:
      return {};
  }
}

private async getRawAssets(component: ComponentXO) {
  const additionalHeaders = getAdditionalHeaders(component.format);

  const assets = await Promise.all(
    component.assets?.map(
      async asset =>
        await this.getRawAsset(asset.downloadUrl, additionalHeaders),
      // Create a dummy promise to avoid Promise.all() from failing
    ) ?? [new Promise<null>(() => null)],
  );

  return assets;
}
```

Reply:
Makes sense, I was thinking it was better to wait to see if Docker is a special case, but it does look ugly on its own. Will update.